final_task/rss_parser/Output_functions.py

```diff
@@ -0,0 +1,227 @@
+                    else:
+                        pack_of_images_links.append(num)
+    logging.info("Got some images!")
+    return pack_of_images_links
+
+
+def getting_alt_text(the_feed, num_of_news):
+    """
+    getting alternative text for images
+    :param the_feed:
+    :param num_of_news:
+    :return:
+    """
+    pack_of_alts = []
+    logging.info("Getting alternative text for images!")
+    for num, item in enumerate(the_feed.entries):
+        if num == getting_num_of_news(the_feed, num_of_news):  # ограничение количества отображаемых новостей
+            break
+        soup = BeautifulSoup(str(item), "lxml")
+        for img in soup.find_all('img'):
+            if img is None:
+                pack_of_alts.append(num)
+            else:
+                try:
+                    pack_of_alts.append(img['alt'])
+                except KeyError:
+                    logging.warning("Solving problems with alternative text!")
+                    pack_of_alts.append(num)
+    logging.info("Got some alternative text.")
+    return pack_of_alts
+
+
+def getting_pack_of_news(the_feed, num_of_news=None):
+    """
+    Creating full novelty
+    1. Adding list of images, correct it if some duplicates are there
+    2. Adding list of alternative texts, correct it if some duplicates are there
+    3. Some problems can occur if there are no alternative text, solving them with changing list of alts
+    :param the_feed:
+    :param num_of_news:
+    :return:
+    """
+    if num_of_news > len(the_feed.entries):
+        print("You want to get more news than exist.")
+        print("Printing all news from the source!")
+
+    print("------------------------")
+    print("Source: ", printing_title(the_feed))
+    pack_of_news = []
+    pack_of_images_links = getting_images_links(the_feed, num_of_news)
+    pack_of_alts = getting_alt_text(the_feed, num_of_news)
+    # corrected_pack_of_images_links = [el for el, _ in groupby(pack_of_images_links)]
+    corrected_pack_of_alts = [el for el, _ in groupby(pack_of_alts)]
+
+    for num, item in enumerate(the_feed.entries):
+        if num == num_of_news:  # ограничение количества отображаемых новостей
+            break
+        pack_of_news.append(getting_novelty(item, num, pack_of_images_links, corrected_pack_of_alts)) # занесение класса Новость в пак новостей
+    return pack_of_news
+
+
+def getting_novelty(item, number, corrected_pack_of_images_links, corrected_pack_of_alts):
+    """
+    It was really difficult to get images because of variety of templates how sites leave link for image
+    Here we create object of Novelty class and fill it with our title, description and etc.
+    Then if some problems with images or alt.text occur we use
+    :param item:
+    :param number:
+    :param corrected_pack_of_images_links:
+    :param corrected_pack_of_alts:
+    :return:
+    """
+
+    try:
+        novelty = Novelty(number + 1, item.get('title', ''), item.get('published', ''),
```

Comment:
In such cases it's way more readable to split parameters one-by-row.
For example:
```
Novelty(
   number + 1,
   item.get('title', ''),
   item.get('published', '')
...
)
```
Same below

Reply:
Fixed.